server/service/src/purchase_order/generate.rs

```diff
@@ -0,0 +1,70 @@
+use repository::{
+    EqualFilter, ItemRowRepository, NumberRowType, PurchaseOrderLineRow, PurchaseOrderRow,
+    RepositoryError, StockOnHandFilter, StockOnHandRepository,
+};
+use util::uuid::uuid;
+
+use crate::number::next_number;
+use crate::service_provider::ServiceContext;
+use crate::store_preference::get_store_preferences;
+
+pub fn generate_empty_purchase_order_lines(
+    ctx: &ServiceContext,
+    purchase_order_row: &PurchaseOrderRow,
+    item_ids: Vec<String>,
+) -> Result<Vec<PurchaseOrderLineRow>, RepositoryError> {
+    let mut result: Vec<PurchaseOrderLineRow> = Vec::new();
+    let store_preferences = get_store_preferences(&ctx.connection, &purchase_order_row.store_id)?;
+
+    let mut line_number = next_number(
+        &ctx.connection,
+        &NumberRowType::PurchaseOrderLine(purchase_order_row.id.clone()),
+        &purchase_order_row.store_id,
+    )?;
+
+    let stocks_on_hand = StockOnHandRepository::new(&ctx.connection).query(Some(
+        StockOnHandFilter::new()
+            .item_id(EqualFilter::equal_any(item_ids.clone()))
+            .store_id(EqualFilter::equal_to(&purchase_order_row.store_id.clone())),
+    ))?;
+
+    item_ids.into_iter().for_each(|item_id| {
+        match ItemRowRepository::new(&ctx.connection).find_active_by_id(&item_id) {
+            Ok(Some(item)) => {
+                let default_pack_size = match store_preferences.pack_to_one {
```

Comment:
I think we should ignore this preference in this context. We use that pack_size_to_one pref to simplify workflows on tablets mainly. If actually ordering stock I think we'd want to specific a real packsize...

Great context thinking of this though!